src/app/App.tsx

```diff
@@ -98,6 +100,30 @@ export class App extends React.Component<Props> {
 
     Linking.addEventListener('url', this.handleOpenURL)
 
+    if (FIREBASE_ENABLED) {
+      this.dynamicLinksRemoveListener = dynamicLinks().onLink(({ url }) =>
+        this.handleOpenURL({ url })
+      )
+
+      if (Platform.OS === 'ios') {
+        const firebaseUrl = await dynamicLinks().getInitialLink()
+
+        if (firebaseUrl) {
+          await this.handleOpenURL({ url: firebaseUrl.url })
+        }
+      }
+    }
+
+    // On Android, initial deep links are picked up by CleverTap - even if they were created by Firebase DynamicLinks.
+    // Breaking out here on Android avoids events being tracked multiple times.
+    if (Platform.OS === 'ios') {
```

Comment:
is it possible to move this inside the `if (FIREBASE_ENABLED)` block above? it seems like doing this would fix the e2e tests (i just tried on my local)